middleware/clean_path.go

```diff
@@ -12,17 +12,19 @@ import (
 func CleanPath(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		rctx := chi.RouteContext(r.Context())
-
-		routePath := rctx.RoutePath
+		
+		var routePath string
+		if rctx != nil {
+			routePath = rctx.RoutePath
+		}
 		if routePath == "" {
 			if r.URL.RawPath != "" {
 				routePath = r.URL.RawPath
 			} else {
 				routePath = r.URL.Path
 			}
-			rctx.RoutePath = path.Clean(routePath)
 		}
-
+		rctx.RoutePath = path.Clean(routePath)
```

Comment:
wont it panic on this line if rctx is nil..? and this PR changes the logic of the method as well, calling path.Clean() more often then previously